### PR TITLE
85290 paysf uos 12 month duration

### PR DESCRIPTION
**Pull request by sakshi15525**
1. Added code to remove paysf records with 12 months duration from UoS updated event.
2. Added code to remove paysf records with 12 months duration from price change event.
3. Unit test cases added.

### CODE REVIEW

File: src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs [hunk 1]
```diff
@@ -31,80 +32,82 @@ public List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceInformatio
                     foreach (var priceInformation in unitPriceInformationList)
                     {
                         var isUnitOfSalePriceExists = unitsOfSalePriceList.Any(x => x.UnitName.Contains(priceInformation.ProductName));
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs:34]: isUnitOfSalePriceExists can be defined as bool

File: src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs [hunk 2]
```diff
@@ -122,72 +122,74 @@ private List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceChange> p
                 List<Price> priceList = new();
 
                 var isUnitOfSalePriceExists = unitsOfSalePriceList.Any(x => x.UnitName.Contains(priceInformation.ProductName));
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs:124]: same here, isUnitOfSalePriceExists define as boolean here

File: src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs [hunk 3]
```diff
@@ -122,72 +122,74 @@ private List<UnitsOfSalePrices> MapAndBuildUnitsOfSalePrices(List<PriceChange> p
...
+                                DateTimeOffset futureDate = GetDate(priceInformation.FutureDate, priceInformation.FutureTime);
+                                Price futurePrice = BuildPricePayload(priceInformation.Duration, priceInformation.FuturePrice, futureDate, priceInformation.FutureCurr);
+                                existingUnitOfSalePrice.Price.Add(futurePrice);
+                            }
                         }
-                    }
+                    } 
                 }
             }
             return unitsOfSalePriceList;
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs:195]: make this method as static

File: tests/UKHO.ERPFacade.Common.UnitTests/Services/ERPFacadeServiceTests.cs [hunk 4]
```diff
@@ -55,6 +55,9 @@ public void WhenUnitOfSaleDetailIsPassed_ThenReturnsUnitsOfSalePrices()
          && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "PriceInformation is missing for {UnitName} in price information payload received from SAP ").MustHaveHappened();
 
             result.Should().BeOfType<List<UnitsOfSalePrices>>();
+
+            var dd = result.Where(x => x.UnitName == "PAYSF").FirstOrDefault();
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.Common.UnitTests/Services/ERPFacadeServiceTests.cs:59]: Kindly remove this and add assertions for PAYSF result and total count

File: tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.PublishPriceChange.WebJob.UnitTests/Services/SlicingPublishingServiceTests.cs:1]: Kindly sort usings

File: tests/UKHO.ERPFacade.Common.UnitTests/Services/ERPFacadeServiceTests.cs
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.Common.UnitTests/Services/ERPFacadeServiceTests.cs:1]: Kindly sort usings for this file

File: src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs
Comment (Shirin14926) [inline src/UKHO.ERPFacade.PublishPriceChange.WebJob/Services/SlicingPublishingService.cs:1]: Kindly sort using namespace

File: src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs
Comment (Shirin14926) [inline src/UKHO.ERPFacade.Common/Services/ERPFacadeService.cs:1]: Remove unused namespace and sort using namespace  please

**Review by Shirin14926 — COMMENTED** [review]
Few Minor comment please

**Review by Shirin14926 — APPROVED** [review]
LGTM .  Thanks @sakshi15525 for making these changes.

**Review by RaviK2233 — APPROVED** [review]
Reviewed FT and looks good.

**Pull request merged**